### ISSUE TITLE
HTML only sometimes renders as HTML

### ISSUE THREAD

**Issue by yurivish** [JuliaLang/julia#25172]
```
julia> struct X x end

julia> Base.show(io::IO, ::MIME"text/html", x::X) = show(io, "Showing as html")

julia> Base.show(io::IO, ::MIME"text/plain", x::X) = show(io, "Showing as text")

julia> x = X("Hi")
"Showing as text"

julia> show(STDOUT, MIME"text/html"(), HTML(x))
X("Hi")

julia> show(STDOUT, MIME"text/html"(), HTML(x, x))
"Showing as html""Showing as html"
```

The last two lines hit different methods of HTML. 

The first calls the constructor: https://github.com/JuliaLang/julia/blob/master/base/docs/utils.jl#L24
The second calls the varag function here: https://github.com/JuliaLang/julia/blob/master/base/docs/utils.jl#L27

The `show` methods immediately below were written in a way that appears to anticipate this behavior. But I find it quite confusing — shouldn't the content passed to the constructor _always_ be shown as "text/html", and not plain text or printed in the single-argument case?

**Comment by yurivish** [comment]
Bump, in case fixing this would constitute a breaking change from 1.0. (It might be considered a bugfix, and deferred until 1.x?)

**Comment by JeffBezanson** [comment]
I agree that type is a bit odd --- it's confusing that it wraps either some content, or a function (?)

It seems that the purpose of this type is to wrap a string in order to declare that it *is* HTML, and so can be used directly when HTML is the requested MIME type. But then the varargs function does something very different.

@MikeInnes What can be done here? Should we change the `show(io, MIME"text/html"()` call to call `print` instead?
